tests/WP_Ultimo/Models/Membership_Test.php

```
@@ -36,267 +36,1343 @@ class Membership_Test extends \WP_UnitTestCase {
-		// Save the membership with updates
-		$result = $this->membership->save();
+		$this->assertSame(Membership_Status::CANCELLED, $this->membership->get_status());
+		$this->assertNotEmpty($this->membership->get_date_cancellation());
+	}
+
+	/**
+	 * Test cancel with already cancelled membership does nothing.
+	 */
+	public function test_cancel_already_cancelled(): void {
+		$this->membership->set_status(Membership_Status::CANCELLED);
+		$old_date = $this->membership->get_date_cancellation();
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$old_date'. (undefined)

(UnusedLocalVariable)